platform/core/src/services/DisplaySetService/DisplaySetService.ts

```diff
@@ -236,34 +269,70 @@ export default class DisplaySetService extends PubSubService {
         );
 
         if (displaySets.length) {
-          this._addActiveDisplaySets(displaySets);
-        } else {
-          displaySets = handler.getDisplaySetsFromSeries(instances);
+          instances = filterInstances(instances, displaySets);
+          for (const ds of displaySets) {
```

Comment:
need more comment what is being done here, you had great explanation in the pr description